### ISSUE TITLE
error during conda installation

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#556]
We should better handle errors if conda fails:

~~~
=============================
SPINAL CORD TOOLBOX INSTALLER
Modified: 2015-10-03
=============================
Checking if administrator rights are needed for installation...
  No sudo needed for adding elements.

Check if spinalcordtoolbox is already installed (if so, delete it)...
  No sudo needed for removing SCT.
>> rm -rf /Users/julien/spinalcordtoolbox
rm -rf /Users/julien/spinalcordtoolbox
Create folder: /Users/julien/spinalcordtoolbox ...
>> mkdir /Users/julien/spinalcordtoolbox
mkdir /Users/julien/spinalcordtoolbox
Fetch version of the Spinal Cord Toolbox...
  Version: 2.1_beta2
Checking for connection and SCT version online...
          
Copy Spinal Cord Toolbox on your computer...
>> cp -r spinalcordtoolbox/* /Users/julien/spinalcordtoolbox
cp -r spinalcordtoolbox/* /Users/julien/spinalcordtoolbox

Edit .bashrc...
  Deleting previous SCT entries in .bashrc
>> awk '!/SCT_DIR|SPINALCORDTOOLBOX|ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS/' ~/.bashrc > .bashrc_temp && > ~/.bashrc && cat .bashrc_temp >> ~/.bashrc && rm .bashrc_temp
awk '!/SCT_DIR|SPINALCORDTOOLBOX|ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS/' ~/.bashrc > .bashrc_temp && > ~/.bashrc && cat .bashrc_temp >> ~/.bashrc && rm .bashrc_temp
  Deleting previous SCT entries in .bash_profile
>> awk '!/SCT_DIR|SPINALCORDTOOLBOX|ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS/' ~/.bash_profile > .bash_profile_temp && > ~/.bash_profile && cat .bash_profile_temp >> ~/.bash_profile && rm .bash_profile_temp
awk '!/SCT_DIR|SPINALCORDTOOLBOX|ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS/' ~/.bash_profile > .bash_profile_temp && > ~/.bash_profile && cat .bash_profile_temp >> ~/.bash_profile && rm .bash_profile_temp

.bashrc seems to be called in .bash_profile
>> . ~/.bashrc
. ~/.bashrc

Installing dependences... Depending on your internet connection, this step may take several minutes.
>> bash requirements.sh
bash requirements.sh
--------------------------
INSTALLER FOR REQUIREMENTS
--------------------------
* CONDA *
Fetching package metadata: ....
Solving package specifications: .
Package plan for installation in environment /Users/julien/miniconda:

The following packages will be downloaded:

package                    |            build
---------------------------|-----------------
jpeg-8d                    |                1         137 KB
libpng-1.6.17              |                0         184 KB
freetype-2.5.2             |                2         692 KB
libtiff-4.0.2              |                1         270 KB
qt-4.8.6                   |                3        39.0 MB
cython-0.23                |           py27_0         2.3 MB
decorator-4.0.4            |           py27_0          11 KB
fastcache-1.0.2            |           py27_0          23 KB
networkx-1.10              |           py27_0         1.1 MB
numpy-1.9.2                |           py27_2         2.9 MB
pillow-2.9.0               |           py27_0         465 KB
pyparsing-2.0.3            |           py27_0          63 KB
python-2.7.10              |                1         9.9 MB
pytz-2015.6                |           py27_0         173 KB
scipy-0.16.0               |       np19py27_1        12.5 MB
setuptools-18.3.2          |           py27_0         342 KB
sip-4.16.5                 |           py27_0         234 KB
six-1.9.0                  |           py27_0          16 KB
sympy-0.7.6                |           py27_0         6.2 MB
wheel-0.26.0               |           py27_1          76 KB
conda-3.18.1               |           py27_0         172 KB
pip-7.1.2                  |           py27_0         1.4 MB
pyqt-4.11.3                |           py27_0         4.0 MB
python-dateutil-2.4.2      |           py27_0         219 KB
matplotlib-1.4.3           |       np19py27_3        40.8 MB
scikit-image-0.11.3        |       np19py27_0        16.1 MB
------------------------------------------------------------
Total:       139.1 MB

The following NEW packages will be INSTALLED:

cython:          0.23-py27_0
decorator:       4.0.4-py27_0
fastcache:       1.0.2-py27_0
freetype:        2.5.2-2
jpeg:            8d-1
libpng:          1.6.17-0
libtiff:         4.0.2-1
matplotlib:      1.4.3-np19py27_3
networkx:        1.10-py27_0
numpy:           1.9.2-py27_2
pillow:          2.9.0-py27_0
pip:             7.1.2-py27_0
pyparsing:       2.0.3-py27_0
pyqt:            4.11.3-py27_0
python-dateutil: 2.4.2-py27_0
pytz:            2015.6-py27_0
qt:              4.8.6-3
scikit-image:    0.11.3-np19py27_0
scipy:           0.16.0-np19py27_1
setuptools:      18.3.2-py27_0
sip:             4.16.5-py27_0
six:             1.9.0-py27_0
sympy:           0.7.6-py27_0
wheel:           0.26.0-py27_1

The following packages will be UPDATED:

conda:           3.16.0-py27_0 --> 3.18.1-py27_0
python:          2.7.10-0      --> 2.7.10-1

Fetching packages ...
jpeg-8d-1.tar. 100% |###############################| Time: 0:00:00 759.76 kB/s
libpng-1.6.17- 100% |###############################| Time: 0:00:00 348.92 kB/s
freetype-2.5.2 100% |###############################| Time: 0:00:09  76.99 kB/s
libtiff-4.0.2- 100% |###############################| Time: 0:00:00 417.98 kB/s
qt-4.8.6-3.tar 100% |###############################| Time: 0:01:18 523.41 kB/s
cython-0.23-py 100% |###############################| Time: 0:00:05 445.35 kB/s
decorator-4.0. 100% |###############################| Time: 0:00:00   9.01 MB/s
fastcache-1.0. 100% |###############################| Time: 0:00:00 724.30 kB/s
networkx-1.10- 100% |###############################| Time: 0:00:14  78.80 kB/s
numpy-1.9.2-py 100% |###############################| Time: 0:00:16 183.81 kB/s
pillow-2.9.0-p 100% |###############################| Time: 0:00:08  56.45 kB/s
pyparsing-2.0. 100% |###############################| Time: 0:00:00 309.49 kB/s
python-2.7.10- 100% |###############################| Time: 0:00:13 785.96 kB/s
pytz-2015.6-py 100% |###############################| Time: 0:00:00 178.53 kB/s
scipy-0.16.0-n 100% |###############################| Time: 0:00:41 318.76 kB/s
setuptools-18. 100% |###############################| Time: 0:00:00 787.91 kB/s
sip-4.16.5-py2 100% |###############################| Time: 0:00:00 500.42 kB/s
six-1.9.0-py27 100% |###############################| Time: 0:00:00 454.91 kB/s
sympy-0.7.6-py 100% |###############################| Time: 0:00:03   1.81 MB/s
wheel-0.26.0-p 100% |###############################| Time: 0:00:00 773.84 kB/s
conda-3.18.1-p 100% |###############################| Time: 0:00:00   1.61 MB/s
pip-7.1.2-py27 100% |###############################| Time: 0:00:00   1.71 MB/s
pyqt-4.11.3-py 100% |###############################| Time: 0:00:01   2.73 MB/s
python-dateuti 100% |###############################| Time: 0:00:00   1.46 MB/s
matplotlib-1.4 100% |###############################| Time: 0:00:16   2.59 MB/s
scikit-image-0 100% |###############################| Time: 0:00:05   2.82 MB/s
Extracting packages ...
An unexpected error has occurred, please consider sending the##          |  80%
following traceback to the conda GitHub issue tracker at:

https://github.com/conda/conda/issues

Include the output of the command 'conda info' in your report.


Traceback (most recent call last):
File "/Users/julien/miniconda/bin/conda", line 5, in <module>
sys.exit(main())
File "/Users/julien/miniconda/lib/python2.7/site-packages/conda/cli/main.py", line 194, in main
args_func(args, p)
File "/Users/julien/miniconda/lib/python2.7/site-packages/conda/cli/main.py", line 201, in args_func
args.func(args, p)
File "/Users/julien/miniconda/lib/python2.7/site-packages/conda/cli/main_install.py", line 45, in execute
install.install(args, parser, 'install')
File "/Users/julien/miniconda/lib/python2.7/site-packages/conda/cli/install.py", line 415, in install
plan.execute_actions(actions, index, verbose=not args.quiet)
File "/Users/julien/miniconda/lib/python2.7/site-packages/conda/plan.py", line 532, in execute_actions
inst.execute_instructions(plan, index, verbose)
File "/Users/julien/miniconda/lib/python2.7/site-packages/conda/instructions.py", line 148, in execute_instructions
cmd(state, arg)
File "/Users/julien/miniconda/lib/python2.7/site-packages/conda/instructions.py", line 62, in EXTRACT_CMD
install.extract(config.pkgs_dirs[0], arg)
File "/Users/julien/miniconda/lib/python2.7/site-packages/conda/install.py", line 455, in extract
t.extractall(path=path)
File "/Users/julien/miniconda/lib/python2.7/tarfile.py", line 2072, in extractall
self.extract(tarinfo, path)
File "/Users/julien/miniconda/lib/python2.7/tarfile.py", line 2109, in extract
self._extract_member(tarinfo, os.path.join(path, tarinfo.name))
File "/Users/julien/miniconda/lib/python2.7/tarfile.py", line 2185, in _extract_member
self.makefile(tarinfo, targetpath)
File "/Users/julien/miniconda/lib/python2.7/tarfile.py", line 2225, in makefile
with bltn_open(targetpath, "wb") as target:
IOError: [Errno 28] No space left on device: u'/Users/julien/miniconda/pkgs/pip-7.1.2-py27_0/lib/python2.7/site-packages/pip/commands/wheel.py'
ERROR: Pip is not installed. Cannot install dependences.

Create links to python scripts...
>> /Users/julien/spinalcordtoolbox/install/create_links.sh -a
/Users/julien/spinalcordtoolbox/install/create_links.sh -a

Checking for available patches...
  No connexion or no patch available for this version of the toolbox.

Compile external packages...
>> python compile_external.py
python compile_external.py
Done! Open a new Terminal window to load environment variables.

Check if other dependent software are installed...
>> sct_check_dependences
sct_check_dependences
/bin/sh: sct_check_dependences: command not found
Traceback (most recent call last):
  File "./installer.py", line 892, in <module>
    Installer()
  File "./installer.py", line 833, in __init__
    status, output = runProcess(cmd)
TypeError: 'NoneType' object is not iterable
~~~

**Comment by jcohenadad** [comment]
fixed by using ``commands.getstatusoutput`` instead of ``runProcess``.

Maybe there was a problem in the way ``runProcess`` was implemented.  If so, it would be good to fix it (used at other instances).

**Comment by jcohenadad** [comment]
we should clean the installer, to deal with exceptions everytime runProcess is called.

**Comment by jcohenadad** [comment]
fixed in 914debeac543432439a7aec173351e09db0a1520
